mypy/semanal.py

```diff
@@ -3505,7 +3509,8 @@ def unwrap_final(self, s: AssignmentStmt) -> bool:
         if self.loop_depth[-1] > 0:
             self.fail("Cannot use Final inside a loop", s)
         if self.type and self.type.is_protocol:
-            self.msg.protocol_members_cant_be_final(s)
+            if self.is_class_scope():
```

Comment:
(at least as I'm reading it, this is the key change and hence my question about line 1623)